cpukit/score/src/schedulerstrongapa.c

```diff
@@ -3,339 +3,820 @@
+
+  insert_priority = _Scheduler_SMP_Node_priority( ready_to_scheduled );
   insert_priority = SCHEDULER_PRIORITY_APPEND( insert_priority );
-  _Chain_Insert_ordered_unprotected(
-    &self->Base.Scheduled,
-    &node->Base.Base.Node.Chain,
-    &insert_priority,
-    _Scheduler_SMP_Priority_less_equal
+  _Scheduler_SMP_Insert_scheduled(
+    context,
+    ready_to_scheduled,
+    insert_priority
   );
 }
 
-static void _Scheduler_strong_APA_Insert_ready(
+/*
+ * Implement the BFS Algorithm for task departure to get the highest ready task
+ * for a particular CPU, returns the highest ready Scheduler_Node
+ * Scheduler_Node filter here pointst to the victim node that is blocked
+ * resulting which this function is called.
+ */
+static inline Scheduler_Node *_Scheduler_strong_APA_Get_highest_ready(
   Scheduler_Context *context,
-  Scheduler_Node    *node_base,
-  Priority_Control   insert_priority
+  Scheduler_Node    *filter
 )
 {
   Scheduler_strong_APA_Context *self;
+  Per_CPU_Control              *filter_cpu;
   Scheduler_strong_APA_Node    *node;
+  Scheduler_Node               *highest_ready;
+  Scheduler_Node               *curr_node;
+  Scheduler_Node               *next_node;
+  Scheduler_strong_APA_CPU     *CPU;
+  uint32_t	                front;
+  uint32_t	                rear;
+  uint32_t	                cpu_max;
+  uint32_t             	        cpu_index;
 
   self = _Scheduler_strong_APA_Get_self( context );
-  node = _Scheduler_strong_APA_Node_downcast( node_base );
+  /* Denotes front and rear of the queue */
+  front = 0;
+  rear = -1;
 
-  if ( SCHEDULER_PRIORITY_IS_APPEND( insert_priority ) ) {
-    _Scheduler_priority_Ready_queue_enqueue(
-      &node->Base.Base.Node.Chain,
-      &node->Ready_queue,
-      &self->Bit_map
-    );
-  } else {
-    _Scheduler_priority_Ready_queue_enqueue_first(
-      &node->Base.Base.Node.Chain,
-      &node->Ready_queue,
-      &self->Bit_map
-    );
+  filter_cpu = _Thread_Get_CPU( filter->user );
+  CPU = self->CPU;
+  cpu_max = _SMP_Get_processor_maximum();
+
+  for ( cpu_index = 0 ; cpu_index < cpu_max ; ++cpu_index ) {
+    CPU[ cpu_index ].visited = false;
   }
+
+  rear = rear + 1;
+  CPU[ rear ].cpu = filter_cpu;
+  CPU[ _Per_CPU_Get_index( filter_cpu ) ].visited = true;
+
+  highest_ready = _Scheduler_strong_APA_Find_highest_ready(
+                    self,
+                    front,
+                    rear
+                  );
+
+  if ( highest_ready != filter ) {
+    /*
+     * Backtrack on the path from
+     * filter_cpu to highest_ready, shifting along every task.
+     */
+
+    node = _Scheduler_strong_APA_Node_downcast( highest_ready );
+    /*
+     * Highest ready is not just directly reachable from the victim cpu
+     * So there is need for task shifting .
+     */
+    while ( node->cpu_to_preempt !=  filter_cpu ) {
+      curr_node = &node->Base.Base;
+      next_node = 
+          _Scheduler_strong_APA_Get_scheduled( self, node->cpu_to_preempt );
```

Comment:
This doesn't look like the preferred way to break lines, I think
```
next_node = _Scheduler_strong_APA_Get_scheduled(
    self,
    node->cpu_to_preempt
 );
```

Reply:
Changed to
next_node = _Scheduler_strong_APA_Get_scheduled(
                       self,
                       node->cpu_to_preempt
                      );
Similar to 
  highest_ready = _Scheduler_strong_APA_Find_highest_ready(
                    self,
                    front,
                    rear
                  );

a few lines above. Is this correct? There's no clear indication in https://docs.rtems.org/branches/master/eng/coding-formatting.html#eighty-character-line-limit, but this looks better ig.

Reply:
yes. you use backticks ` to create code snippets

Reply:
Thanks. Yeah, I used it, but the indentations did not look as expected.